src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -216,10 +208,10 @@ else if (shouldCheckLastRcurly) {
                 violation = MSG_KEY_LINE_ALONE;
             }
         }
-        else if (shouldBeOnSameLine(bracePolicy, details)) {
+        else if (shouldBeOnSameLine(option, details)) {
             violation = MSG_KEY_LINE_SAME;
         }
-        else if (shouldBeAloneOnLine(bracePolicy, details)) {
+        else if (shouldBeAloneOnLine(option, details)) {
             violation = MSG_KEY_LINE_ALONE;
         }
         else if (shouldStartLine && !isOnStartOfLine(details, targetSourceLine)) {
```

Comment:
@rnveach 
That will work. I fetched @sagar-shah94's branch and made changes in my local repo. Diff reports will be OK. Checkstyle will not rise CC violation.

@sagar-shah94 
Please, make changes suggested by @rnveach .